File: qb-engineer-ui/src/app/features/vendors/models/create-vendor-request.model.ts
export interface CreateVendorRequest {
  companyName: string;
  contactName?: string;
  email?: string;
  phone?: string;
  address?: string;
  city?: string;
  state?: string;
  zipCode?: string;
  country?: string;
  paymentTerms?: string;
  notes?: string;
}

File: qb-engineer-server/qb-engineer.core/Models/CreateVendorRequestModel.cs
namespace QBEngineer.Core.Models;

public record CreateVendorRequestModel(
    string CompanyName,
    string? ContactName,
    string? Email,
    string? Phone,
    string? Address,
    string? City,
    string? State,
    string? ZipCode,
    string? Country,
    string? PaymentTerms,
    string? Notes);
